### ISSUE TITLE
[feature] Support multiple remotes on install command

### ISSUE THREAD

**Issue by ghost** [conan-io/conan#7998]
- [] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).

Currently, there is a syntax to **search** multiple (all) remotes, like so:
`conan search -r all  gtest/1.10.0`

There is not, however, a way to install packages from multiple/all remotes:
```bash
# Does not work
conan install gtest/1.10.0 -r=all
# Or, does not work as well..
conan install gtest/1.10.0 -r=remote_1 -r=remote_2
```

Is there a reason for that? Would greatly appreciate this.
Thanks,

PS:
The [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md) link is broken.

**Comment by memsharded** [comment]
Hi @tomerlevv 

Installing from all the remotes is the default already, no need to specify ``-r=all``.

For installing from more than 1 remote, there are some quirks in the assumptions that are made when a remote is explicitly passed as an argument that could make implementing this a bit inconvenient and risk of introducing subtle bugs.

I would like to implement this, but I don't think we have the capacity to do it right before Conan 2.0, so I propose implementing this feature but not now.

In the meantime, have you tried the Artifactory virtual repositories? You could perfectly define a virtual repository there that aggregates both your remote1 and remote2, and then you will be able to specify ``-r=virtual``, and the effect will be that it will fetch from both remote 1 and remote2.

**Comment by ghost** [comment]
Hey @memsharded , thanks for the answer. I will look at virtual repositories. 
But I can't see the behavior you're discussing regarding that conan looks at all the remotes when not specifying `-r`. 
Example, without specifying a `-r`flag, conan can't find my custom arm-based gtest, looks only in conan-center:
```
container@c51c1bae97d9:/sl$ rm -rf ~/.conan/data
container@c51c1bae97d9:/sl$ conan install gtest/1.10.0@ -pr ./.conan/profiles/arm 
Configuration:
[settings]
arch=armv7hf
arch_build=x86_64
build_type=Release
compiler=gcc
compiler.libcxx=libstdc++11
compiler.version=6
os=Linux
os_build=Linux
[options]
[build_requires]
[env]
AR=arm-linux-gnueabihf-ar
CC=arm-linux-gnueabihf-gcc
CXX=arm-linux-gnueabihf-g++
LD=arm-linux-gnueabihf-ld
RANLIB=arm-linux-gnueabihf-ranlib
STRIP=arm-linux-gnueabihf-strip
gtest/1.10.0: Not found in local cache, looking in remotes...
gtest/1.10.0: Trying with 'conan-center'...
Downloading conanmanifest.txt completed [0.39k]                                          
Downloading conanfile.py completed [4.74k]                                               
Downloading conan_export.tgz completed [0.30k]                                           
Decompressing conan_export.tgz completed [0.00k]                                         
gtest/1.10.0: Downloaded recipe revision 0
Installing package: gtest/1.10.0
Requirements
    gtest/1.10.0 from 'conan-center' - Downloaded
Packages
    gtest/1.10.0:f2bcd6b363c34bd1e44869a1c8cd8df32379d528 - Missing

Cross-build from 'Linux:x86_64' to 'Linux:armv7hf'
Installing (downloading, building) binaries...
ERROR: Missing binary: gtest/1.10.0:f2bcd6b363c34bd1e44869a1c8cd8df32379d528

gtest/1.10.0: WARN: Can't find a 'gtest/1.10.0' package for the specified settings, options and dependencies:
- Settings: arch=armv7hf, build_type=Release, compiler=gcc, compiler.libcxx=libstdc++11, compiler.version=6, os=Linux
- Options: build_gmock=True, fPIC=True, hide_symbols=False, no_main=False, shared=False
- Dependencies: 
- Requirements: 
- Package ID: f2bcd6b363c34bd1e44869a1c8cd8df32379d528

ERROR: Missing prebuilt package for 'gtest/1.10.0'
Try to build from sources with '--build=gtest'
Use 'conan search <reference> --table table.html'
Or read 'http://docs.conan.io/en/latest/faq/troubleshooting.html#error-missing-prebuilt-package'
```
But, when specifying custom remote which has that package: 
```
container@c51c1bae97d9:/sl$ conan install gtest/1.10.0@ -pr ./.conan/profiles/arm -r argus
Configuration:
[settings]
arch=armv7hf
arch_build=x86_64
build_type=Release
compiler=gcc
compiler.libcxx=libstdc++11
compiler.version=6
os=Linux
os_build=Linux
[options]
[build_requires]
[env]
AR=arm-linux-gnueabihf-ar
CC=arm-linux-gnueabihf-gcc
CXX=arm-linux-gnueabihf-g++
LD=arm-linux-gnueabihf-ld
RANLIB=arm-linux-gnueabihf-ranlib
STRIP=arm-linux-gnueabihf-strip
gtest/1.10.0: Retrieving from server 'argus' 
gtest/1.10.0: Trying with 'argus'...
Downloading conanmanifest.txt completed [0.39k]                                          
Downloading conanfile.py completed [4.74k]                                               
Downloading conan_export.tgz completed [0.30k]                                           
Decompressing conan_export.tgz completed [0.00k]                                         
gtest/1.10.0: Downloaded recipe revision 0
Installing package: gtest/1.10.0
Requirements
    gtest/1.10.0 from 'argus' - Downloaded
Packages
    gtest/1.10.0:f2bcd6b363c34bd1e44869a1c8cd8df32379d528 - Download

Cross-build from 'Linux:x86_64' to 'Linux:armv7hf'
Installing (downloading, building) binaries...
gtest/1.10.0: Retrieving package f2bcd6b363c34bd1e44869a1c8cd8df32379d528 from remote 'argus' 
Downloading conanmanifest.txt completed [3.64k]                                          
Downloading conaninfo.txt completed [0.74k]                                              
Downloading conan_package.tgz completed [415.67k]                                        
Decompressing conan_package.tgz completed [0.00k]                                        
gtest/1.10.0: Package installed f2bcd6b363c34bd1e44869a1c8cd8df32379d528
gtest/1.10.0: Downloaded package revision 0
```
Am I missing something here?
Thanks,

**Comment by memsharded** [comment]
> Example, without specifying a -rflag, conan can't find my custom arm-based gtest, looks only in conan-center:

It depends on the mode of evaluation of remotes, which depends if using revisions or not. If you haven't enabled revisions, then Conan will look for the first remote when it can find the recipe, and from there it will use only that remote to get binaries. This is done for consistency, if you are not using revisions, you cannot guarantee that the recipe is the same in different remotes, and this can be very problematic.

If you enable revisions, then, when a binary is missing for that specific revision, it can go to other remotes which contain the same revision and look for that binary. Please try this and let us know.

Revisions is a modern feature that was introduced in 1.X, and couldn't be made the default one because it would be breaking. It will be the default mode of operation in Conan 2.0.

**Comment by memsharded** [comment]
Conan 2.0 iterates the remotes for install, this is implemented and released in Conan 2.0-beta. Closing.
